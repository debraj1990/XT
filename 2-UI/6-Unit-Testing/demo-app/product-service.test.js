

let searchProducts = require('./product-service');

expect.addSnapshotSerializer({
    test: (val) => val.id && val.name && val.price,
    print: (val) => `${val.name} ${val.price}`
});

describe('product-service tests', () => {


    // it('return the products that match query', () => {
    //     expect(searchProducts('M')).toEqual(
    //         [{ id: 2, name: 'Mobile', price: 2000, description: 'New ...' },
    //         { id: 3, name: 'Mobile 2.0', price: 3000 },
    //         { id: 4, name: 'Mobile 3.0', price: 4000 },
    //         ]
    //     )
    // })


    it('return the products that match query', () => {
        ['M', 'Mo', 'Lap', 'Cam'].forEach(query => {
            expect(searchProducts(query)).toMatchSnapshot()
        })
    })

    it('return the products that match query v2', () => {
        expect(searchProducts('Lap')).toMatchSnapshot()
    })

    it('return the products that match query v2', () => {
        expect(searchProducts('Lap')).toMatchSnapshot()
    })




})