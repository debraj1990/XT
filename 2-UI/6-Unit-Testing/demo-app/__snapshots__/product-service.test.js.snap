// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product-service tests return the products that match query 1`] = `
Array [
  Mobile 2000,
  Mobile 2.0 3000,
  Mobile 3.0 4000,
  Mobile 4.0 4000,
]
`;

exports[`product-service tests return the products that match query 2`] = `
Array [
  Mobile 2000,
  Mobile 2.0 3000,
  Mobile 3.0 4000,
  Mobile 4.0 4000,
]
`;

exports[`product-service tests return the products that match query 3`] = `
Array [
  Laptop 1000,
]
`;

exports[`product-service tests return the products that match query 4`] = `Array []`;

exports[`product-service tests return the products that match query v2 1`] = `
Array [
  Laptop 1000,
]
`;

exports[`product-service tests return the products that match query v2 2`] = `
Array [
  Laptop 1000,
]
`;
